addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1707,6 +1724,122 @@ export class OdooEditor extends EventTarget {
+            }
+
+            // Create the add buttons.
+            if (direction === 'column') {
+                const addColumnLeft = document.createElement('div');
+                addColumnLeft.textContent = 'Insert left';
+                const addIconLeft = document.createElement('span');
+                addIconLeft.classList.add('fa', 'fa-plus');
+                addColumnLeft.prepend(addIconLeft);
+                const addColumnRight = document.createElement('div');
+                addColumnRight.textContent = 'Insert right';
+                const addIconRight = document.createElement('span');
+                addIconRight.classList.add('fa', 'fa-plus');
+                addColumnRight.prepend(addIconRight);
+                uiMenu.append(addColumnLeft);
+                uiMenu.append(addColumnRight);
+                this.addDomListener(addColumnLeft, 'click', this._onTableAddColumnLeftClick);
+                this.addDomListener(addColumnRight, 'click', this._onTableAddColumnRightClick);
+            } else {
+                const addRowAbove = document.createElement('div');
+                addRowAbove.textContent = 'Insert above';
+                const addIconAbove = document.createElement('span');
+                addIconAbove.classList.add('fa', 'fa-plus');
+                addRowAbove.prepend(addIconAbove);
+                const addRowBelow = document.createElement('div');
+                addRowBelow.textContent = 'Insert below';
+                const addIconBelow = document.createElement('span');
+                addIconBelow.classList.add('fa', 'fa-plus');
+                addRowBelow.prepend(addIconBelow);
+                uiMenu.append(addRowAbove);
+                uiMenu.append(addRowBelow);
+                this.addDomListener(addRowAbove, 'click', this._onTableAddRowAboveClick);
+                this.addDomListener(addRowBelow, 'click', this._onTableAddRowBelowClick);
+            }
+
+            // Add the delete button.
+            if (direction === 'column') {
+                const deleteColumn = document.createElement('div');
+                deleteColumn.textContent = 'Delete';
+                const removeIcon = document.createElement('span');
+                removeIcon.classList.add('fa', 'fa-trash');
+                deleteColumn.prepend(removeIcon);
+                uiMenu.append(deleteColumn);
+                this.addDomListener(deleteColumn, 'click', this._onTableDeleteColumnClick);
+            } else {
+                const deleteRow = document.createElement('div');
+                deleteRow.textContent = 'Delete';
+                const removeIcon = document.createElement('span');
+                removeIcon.classList.add('fa', 'fa-trash');
+                deleteRow.prepend(removeIcon);
+                uiMenu.append(deleteRow);
+                this.addDomListener(deleteRow, 'click', this._onTableDeleteRowClick);
+            }
```

Comment:
Wow... Couldn't we use template strings for all this ? Or static qweb templates ?

Reply:
Yes I'll rewrite this to use `domParser` and template strings.